sdk/translation/Azure.AI.Translation.Document/src/DocumentTranslationModelFactory.cs

```diff
@@ -0,0 +1,160 @@
+        /// <param name="translatedDocumentUri">Sets the <see cref="DocumentStatus.TranslatedDocumentUri"/> property.</param>
+        /// <param name="sourceDocumentUri">Sets the <see cref="DocumentStatus.SourceDocumentUri"/> property.</param>
+        /// <param name="createdOn">Sets the <see cref="DocumentStatus.CreatedOn"/> property.</param>
+        /// <param name="lastModified">Sets the <see cref="DocumentStatus.LastModified"/> property.</param>
+        /// <param name="status">Sets the <see cref="DocumentStatus.Status"/> property.</param>
+        /// <param name="translatedTo">Sets the <see cref="DocumentStatus.TranslatedTo"/> property.</param>
+        /// <param name="progress">Sets the <see cref="DocumentStatus.Progress"/> property.</param>
+        /// <param name="charactersCharged">Sets the <see cref="DocumentStatus.CharactersCharged"/> property.</param>
+        /// <returns>A new instance of <see cref="Document.DocumentStatus"/> for mocking purposes.</returns>
+        public static DocumentStatus DocumentStatus(
+            string id,
+            Uri translatedDocumentUri,
+            Uri sourceDocumentUri,
+            DateTimeOffset createdOn,
+            DateTimeOffset lastModified,
+            DocumentTranslationStatus status,
+            string translatedTo,
+            float progress,
+            long charactersCharged
+            )
+        {
+            return new DocumentStatus(translatedDocumentUri, sourceDocumentUri, createdOn, lastModified, status, translatedTo, default, progress, id, charactersCharged);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="Document.DocumentStatus"/> for mocking purposes.
+        /// </summary>
+        /// <param name="id">Sets the <see cref="DocumentStatus.Id"/> property.</param>
+        /// <param name="sourceDocumentUri">Sets the <see cref="DocumentStatus.SourceDocumentUri"/> property.</param>
+        /// <param name="createdOn">Sets the <see cref="DocumentStatus.CreatedOn"/> property.</param>
+        /// <param name="lastModified">Sets the <see cref="DocumentStatus.LastModified"/> property.</param>
+        /// <param name="status">Sets the <see cref="DocumentStatus.Status"/> property.</param>
+        /// <param name="translatedTo">Sets the <see cref="DocumentStatus.TranslatedTo"/> property.</param>
+        /// <param name="error">Sets the <see cref="DocumentStatus.Error"/> property.</param>
+        /// <param name="progress">Sets the <see cref="DocumentStatus.Progress"/> property.</param>
+        /// <param name="charactersCharged">Sets the <see cref="DocumentStatus.CharactersCharged"/> property.</param>
+        /// <returns>A new instance of <see cref="Document.DocumentStatus"/> for mocking purposes.</returns>
+        public static DocumentStatus DocumentStatus(
+            string id,
+            Uri sourceDocumentUri,
+            DateTimeOffset createdOn,
```

Comment:
I am trying to think about how to make it obvious to the customer that this constructor is for the error scenarios.
Maybe if add `error` here and then we have the other properties `createdOn`, `lastModified`, etc